x-pack/plugins/cases/public/components/user_actions/assignees.tsx

```diff
@@ -5,13 +5,159 @@
  * 2.0.
  */
 
-import type { UserActionBuilder } from './types';
+import { EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { UserProfileWithAvatar } from '@kbn/user-profile-components';
+import React, { memo } from 'react';
+import { Actions, AssigneesUserAction } from '../../../common/api';
+import { getName } from '../user_profiles/display_name';
+import { Assignee } from '../user_profiles/types';
+import { UserToolTip } from '../user_profiles/user_tooltip';
+import { createCommonUpdateUserActionBuilder } from './common';
+import type { UserActionBuilder, UserActionResponse } from './types';
+import * as i18n from './translations';
+import { getUsernameDataTestSubj } from '../user_profiles/data_test_subject';
+
+const FormatListItem: React.FC<{
+  children: React.ReactElement;
+  index: number;
+  listSize: number;
+}> = ({ children, index, listSize }) => {
+  if (shouldAddAnd(index, listSize)) {
+    return (
+      <>
+        {i18n.AND_SPACE}
+        {children}
+      </>
+    );
+  } else if (shouldAddComma(index, listSize)) {
+    return (
+      <>
+        {children}
+        {','}
+      </>
+    );
+  }
+
+  return children;
+};
+FormatListItem.displayName = 'FormatListItem';
+
+const shouldAddComma = (index: number, arrayLength: number) => {
+  return arrayLength > 0 && index !== arrayLength - 1;
+};
+
+const shouldAddAnd = (index: number, arrayLength: number) => {
+  return arrayLength > 1 && index === arrayLength - 1;
+};
+
+const Themselves: React.FC<{
+  index: number;
+  numOfAssigness: number;
+}> = ({ index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <>{i18n.THEMSELVES}</>
+  </FormatListItem>
+);
+Themselves.displayName = 'Themselves';
+
+const AssigneeComponent: React.FC<{
+  assignee: Assignee;
+  index: number;
+  numOfAssigness: number;
+}> = ({ assignee, index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <UserToolTip profile={assignee.profile}>
+      <strong>{getName(assignee.profile?.user)}</strong>
+    </UserToolTip>
+  </FormatListItem>
+);
+AssigneeComponent.displayName = 'Assignee';
+
+interface AssigneesProps {
+  assignees: Assignee[];
+  currentUserProfile?: UserProfileWithAvatar;
+}
+
+const AssigneesComponent = ({ assignees, currentUserProfile }: AssigneesProps) => (
+  <>
+    {assignees.length > 0 && (
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
+        {assignees.map((assignee, index) => {
+          const usernameDataTestSubj = getUsernameDataTestSubj(assignee);
+
+          return (
+            <EuiFlexItem
+              data-test-subj={`ua-assignee-${usernameDataTestSubj}`}
+              grow={false}
+              key={assignee.uid}
+            >
+              <EuiText size="s" className="eui-textBreakWord">
```

Comment:
Here I logged in as Bad Hawk, and assigned that user as well as some other users. In the comment it lists `Bad Hawk` and `themselves` - should only add an item for `themselves`:

<img width="706" alt="image" src="https://user-images.githubusercontent.com/7405507/186674413-1dad1bc0-69ec-4166-8a1e-3d752dac0064.png">

Reply:
I can't reproduce this one a second time! Maybe some stale state?